webapp/loggedin/tableViews/tableView.js

```diff
@@ -1,57 +1,99 @@
+/* eslint react/jsx-props-no-spreading: 0 */
+// TODO: Now TableHeader and TableContent need to pass the dynamic ...rest props to components - All components using TableView must be refactor
 import React, { useEffect } from 'react'
-import { connect } from 'react-redux'
+import PropTypes from 'prop-types'
+import { useDispatch } from 'react-redux'
 
-import * as SurveyState from '@webapp/survey/surveyState'
+import { useOnSurveyCycleUpdate, useSurveyId } from '@webapp/commonComponents/hooks'
 import TableHeader from './components/tableHeader'
 import TableContent from './components/tableContent'
 
-import * as TableViewsState from './tableViewsState'
+import { initListItems, reloadListItems } from './actions'
 
-import { initListItems, fetchListItems } from './actions'
+const TableView = (props) => {
+  const {
+    className,
+    gridTemplateColumns,
+    headerLeftComponent,
+    isRowActive,
+    module,
+    moduleApiUri,
+    noItemsLabelKey,
+    onRowClick,
+    reloadOnSurveyCycleUpdate,
+    restParams,
+    rowComponent,
+    rowHeaderComponent,
+    ...rest
+  } = props
 
-const TableView = props => {
-  const { module, moduleApiUri, restParams, className, initListItems } = props
+  const dispatch = useDispatch()
+  const surveyId = useSurveyId()
+  const apiUri = moduleApiUri || `/api/survey/${surveyId}/${module}`
```

Comment:
should this go into the defaultProps?

Reply:
no, because it uses the survey id. defaultProps are static